src/modules/ethereum/transactions/interpret.ts

```diff
@@ -1,12 +1,222 @@
-import { type Client, client as defaultClient } from 'defi-sdk';
+import { baseToCommon } from 'src/shared/units/convert';
+import type { Asset, AddressAction, Client } from 'defi-sdk';
+import { client as defaultClient } from 'defi-sdk';
 import { rejectAfterDelay } from 'src/shared/rejectAfterDelay';
 import { valueToHex } from 'src/shared/units/valueToHex';
-import type { Chain } from 'src/modules/networks/Chain';
+import { createChain, type Chain } from 'src/modules/networks/Chain';
 import type { MultichainTransaction } from 'src/shared/types/MultichainTransaction';
+import { capitalize } from 'capitalize-ts';
+import { getDecimals } from 'src/modules/networks/asset';
 import type { TypedData } from '../message-signing/TypedData';
 import type { InterpretResponse } from './types';
 import { getGas } from './getGas';
 import type { ChainId } from './ChainId';
+import { getFungibleAsset } from './actionAsset';
+import { convertAssetToFungibleOutline } from './addressAction';
+
+type LegacyInterpretResponse = Omit<InterpretResponse, 'action'> & {
+  action: AddressAction | null;
+};
+
+async function convertToNewInterpretation(
```

Comment:
Temporatey function until backend support new data scheme in the new ZPI endpoint